packages/minimongo/minimongo.js

```diff
@@ -103,7 +103,7 @@ LocalCollection.Cursor = function (collection, selector, options) {
   self.limit = options.limit;
   self.fields = options.fields;
 
-  self._projectionFn = LocalCollection._compileProjection(self.fields || {});
+  self._projectionFn = LocalCollection._compileProjection(self.fields || {}, _.clone);
```

Comment:
Yeah, which is why I'm somewhat concerned about safety. With the shallow clone there are potentially some issues of data ownership if the observer callbacks are mutating the data (although not find/findOne as those do their own deep clone). I haven't thoroughly worked through the reasoning yet.

If nobody sees anything obviously wrong I can try to write an argument that there are no data ownership issues (which might work if there are in fact no data ownership issues)